### PR TITLE
Remove PENDING_EVALUATION as a status, and allow a user to start requ…

### PR DESCRIPTION
**Pull request by mehalshah**
…ired modules before they have taken the evaluation

### CODE REVIEW

**Comment by mehalshah** [comment]
PENDING_EVALUATION is no longer needed as a status so I'm taking it out.

Also, users should be enrolled in default modules as soon as they are enrolled in course units.

File: dashboard/app/models/plc/enrollment_unit_assignment.rb [hunk 1]
```diff
@@ -25,39 +25,41 @@ class Plc::EnrollmentUnitAssignment < ActiveRecord::Base
...
+        module_assignment_for_type(module_type).try(:destroy)
+      end
 
-        end
+      learning_modules.each do |learning_module|
+        enroll_in_module learning_module
       end
-      self.update(status: IN_PROGRESS)
+
+      update(status: IN_PROGRESS)
```
Comment (aoby) [inline dashboard/app/models/plc/enrollment_unit_assignment.rb]: ok that failures aren't caught here? Note line 70 below uses update!

File: dashboard/app/models/plc/enrollment_unit_assignment.rb [hunk 2]
```diff
@@ -68,4 +70,18 @@ def check_for_unit_completion
       update!(status: COMPLETED)
     end
   end
+
+  private
+  def enroll_in_module(learning_module)
+    return unless learning_module.plc_course_unit == plc_course_unit
```
Comment (aoby) [inline dashboard/app/models/plc/enrollment_unit_assignment.rb:76]: Why would this happen? It's a private method called only from line 42 which iterates `plc_course_unit.plc_learning_modules`  And if it does occur, shouldn't we raise an error?

File: dashboard/app/models/plc/enrollment_unit_assignment.rb (hunk 1 above)
Comment (mehalshah) [inline dashboard/app/models/plc/enrollment_unit_assignment.rb]: This should be update! - will change

File: dashboard/app/models/plc/enrollment_unit_assignment.rb (hunk 2 above)
Comment (mehalshah) [inline dashboard/app/models/plc/enrollment_unit_assignment.rb:76]: It's also called in line 59 - which takes a list of learning modules that actually come in from an external call. It's meant to guard against someone submitting an arbitrary list of modules by mucking with the form

Comment (aoby) [inline dashboard/app/models/plc/enrollment_unit_assignment.rb:76]: gotcha. I just wanted to be sure we're ok with silently skipping the irrelevant ones.

**Pull request merged**